File: backend/app/schemas/permission.py
import uuid
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime

class PermissionBase(BaseModel):
    name: str = Field(..., examples=["read_users"])
    description: Optional[str] = Field(None, examples=["Permission to read user data"])

class PermissionCreate(PermissionBase):
    pass

class PermissionUpdate(PermissionBase):
    name: Optional[str] = None
    description: Optional[str] = None

class PermissionRead(PermissionBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True # SQLAlchemyモデルからPydanticモデルへの変換を有効化 (v1)
        # from_attributes = True # Pydantic v2 の場合 

File: study-support-app/src/types/permission.ts
// study-support-app/src/types/permission.ts
export interface PermissionRead {
  id: string; // UUID は string として扱うのが一般的
  name: string;
  description?: string | null;
  created_at: string; // DateTime は string (ISO 8601) として扱う
  updated_at: string; // DateTime は string (ISO 8601) として扱う
}

// 必要に応じて PermissionCreate, PermissionUpdate も定義できますが、
// 今回の UI では直接使用しないため PermissionRead のみ定義します。
// export interface PermissionCreate {
//   name: string;
//   description?: string | null;
// }
// export interface PermissionUpdate {
//   name?: string;
//   description?: string | null;
// } 